nixos/modules/system/boot/systemd.nix

```diff
@@ -992,24 +1011,18 @@ in
       "sysctl.d/50-coredump.conf".source = "${systemd}/example/sysctl.d/50-coredump.conf";
       "sysctl.d/50-default.conf".source = "${systemd}/example/sysctl.d/50-default.conf";
 
-      "tmpfiles.d/00-nixos.conf".text = ''
-        # This file is created automatically and should not be modified.
-        # Please change the option ‘systemd.tmpfiles.rules’ instead.
-
-        ${concatStringsSep "\n" cfg.tmpfiles.rules}
-      '';
-
-      "tmpfiles.d/home.conf".source = "${systemd}/example/tmpfiles.d/home.conf";
-      "tmpfiles.d/journal-nocow.conf".source = "${systemd}/example/tmpfiles.d/journal-nocow.conf";
-      "tmpfiles.d/portables.conf".source = "${systemd}/example/tmpfiles.d/portables.conf";
-      "tmpfiles.d/static-nodes-permissions.conf".source = "${systemd}/example/tmpfiles.d/static-nodes-permissions.conf";
-      "tmpfiles.d/systemd.conf".source = "${systemd}/example/tmpfiles.d/systemd.conf";
-      "tmpfiles.d/systemd-nologin.conf".source = "${systemd}/example/tmpfiles.d/systemd-nologin.conf";
-      "tmpfiles.d/systemd-nspawn.conf".source = "${systemd}/example/tmpfiles.d/systemd-nspawn.conf";
-      "tmpfiles.d/systemd-tmp.conf".source = "${systemd}/example/tmpfiles.d/systemd-tmp.conf";
-      "tmpfiles.d/tmp.conf".source = "${systemd}/example/tmpfiles.d/tmp.conf";
-      "tmpfiles.d/var.conf".source = "${systemd}/example/tmpfiles.d/var.conf";
-      "tmpfiles.d/x11.conf".source = "${systemd}/example/tmpfiles.d/x11.conf";
```

Comment:
Previously I could disable `tmp.conf` (to keep `/tmp` across reboots) with `environment.etc."tmpfiles.d/tmp.conf".text = "# disabled by me";`. This does not work anymore. How shall I achieve that now? May these .source lines be restored? (I do not see the benefit of the new way.)

Reply:
Oof. @ajs124 Can files be overridden with a package that is later in the list?

Reply:
@dasJ No, I don't think so. We could probably do this in nix instead of a derivation, then you could mkForce the option.
@orivej the benefit, like with other similar systemd module options, is that we can use tmpfile files from upstream packages easily.

Reply:
@ajs124 I see the purpose of `systemd.tmpfiles.packages` in general, but I do not see the benefit of the new synthetic packages "systemd-default-tmpfiles" and "nixos-tmpfiles.d" over the old `environment.etc."tmpfiles.d/00-nixos.conf".text = …;  "tmpfiles.d/tmp.conf".source = `.

Reply:
I suppose that using solely `systemd.tmpfiles.packages` for managing `/etc/tmpfiles.d` is good for uniformity, but it looses the well thought out infrastructure for customizing `/etc` via `environment.etc` settings. I think that `systemd.tmpfiles.packages` should be empty by default to allow the user an easy choice between using unmodified upstream tmpfiles.d and custom alternatives.

Reply:
You could also `lib.mkForce` the list to get rid of the default packages.

Reply:
A simple solution would be maintaining a exclude list of `.conf` files, which shouldn't be added to `/etc` in the end.

We could also use `buildEnv` instead of `symlinkJoin`, to allow adding a package with a higher priority which could override existing files?

Reply:
> A simple solution would be maintaining a exclude list of .conf files, which shouldn't be added to /etc in the end.

This sounds good if we would exclude all files in `environment.etc`.

Reply:
> You could also lib.mkForce the list to get rid of the default packages.

This is not acceptable because I want to keep all other systemd tmpfiles.d installed.

Reply:
> This sounds good if we would exclude all files in environment.etc.

Not sure if I understand. `/etc/tmpfiles.d` would still symlink into the nix store, but the code assembling it could have an exclusion list.

Also, what do you think about using `buildEnv`, and its priorities mechanism?

Reply:
I've just opened #96755 to track this.

> Also, what do you think about using buildEnv, and its priorities mechanism?

Sounds like an improvement to me as it doesn't require Nixpkgs patches anymore but it's likely still a pretty verbose solution that isn't that use friendly (though I would be ok with it and I'm not sure if we could easily do better).

I guess a pretty compact snipped to use this would look something like this:
```nix
systemd.tmpfiles.packages = [
  (lib.hiPrio (pkgs.writeTextDir "lib/tmpfiles.d/tmp.conf" ''
    # Disabled
  ''))
];
```

Reply:
@flokli See #96766, I'm using `environment.etc` as the exclusion list.

`buildEnv` seems incompatible with mutable `/etc` and the way `environment.etc` is implemented.